File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/readmodel/ConferenceSlugIndexRepository.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using Conference.Common;
using ConferenceManagement.Domain.Models;
using ConferenceManagement.Domain.Repositories;
using ECommon.Components;
using ECommon.Dapper;

namespace ConferenceManagement.Repositories.Dapper
{
    [Component]
    public class ConferenceSlugIndexRepository : IConferenceSlugIndexRepository
    {
        public void Add(ConferenceSlugIndex index)
        {
            using (var connection = GetConnection())
            {
                connection.Insert(new
                {
                    IndexId = index.IndexId,
                    ConferenceId = index.ConferenceId,
                    Slug = index.Slug
                }, ConfigSettings.ConferenceSlugIndexTable);
            }
        }
        public ConferenceSlugIndex FindSlugIndex(string slug)
        {
            using (var connection = GetConnection())
            {
                var record = connection.QueryList(new { Slug = slug }, ConfigSettings.ConferenceSlugIndexTable).SingleOrDefault();
                if (record != null)
                {
                    var indexId = record.IndexId as string;
                    var conferenceId = (Guid)record.ConferenceId;
                    return new ConferenceSlugIndex(indexId, conferenceId, slug);
                }
                return null;
            }
        }

        private IDbConnection GetConnection()
        {
            return new SqlConnection(ConfigSettings.ConferenceConnectionString);
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/readmodel/ConferenceSlugIndexRepository.java
package com.microsoft.conference.management.readmodel;

import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
import com.microsoft.conference.common.dataobject.ConferenceSlugIndexDO;
import com.microsoft.conference.common.mapper.ConferenceSlugIndexMapper;
import com.microsoft.conference.management.domain.models.ConferenceSlugIndex;
import com.microsoft.conference.management.domain.repositories.IConferenceSlugIndexRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ConferenceSlugIndexRepository implements IConferenceSlugIndexRepository {

    @Autowired
    private ConferenceSlugIndexMapper conferenceSlugIndexMapper;

    @Override
    public void add(ConferenceSlugIndex index) {
        ConferenceSlugIndexDO slugIndexDO = new ConferenceSlugIndexDO();
        slugIndexDO.setConferenceId(index.getConferenceId());
        slugIndexDO.setIndexId(index.getIndexId());
        slugIndexDO.setSlug(index.getSlug());
        conferenceSlugIndexMapper.insert(slugIndexDO);
    }

    @Override
    public ConferenceSlugIndex findSlugIndex(String slug) {
        LambdaQueryWrapper<ConferenceSlugIndexDO> queryWrapper = new LambdaQueryWrapper<>();
        queryWrapper.eq(ConferenceSlugIndexDO::getSlug, slug);
        ConferenceSlugIndexDO slugIndexDO = conferenceSlugIndexMapper.selectOne(queryWrapper);
        return ConferenceConvert.INSTANCE.toDTO(slugIndexDO);
    }
}
